lib/bot.js

```diff
@@ -106,44 +88,55 @@ class Bot {
     return this._safelyRunApiCommand({method: 'read', options: {channel, peek, unreadOnly}})
   }
 
-  // --------------------------------------------------------------------------
-
   chatDelete(options: {channel: ChatChannel, messageId: number}): Promise<any> {
     const {channel, messageId} = options
     return this._safelyRunApiCommand({method: 'delete', options: {channel, messageId}})
   }
 
-  // --------------------------------------------------------------------------
-
   myInfo(): ?DeviceUsernamePair {
     return this._dPair
   }
 
-  // --------------------------------------------------------------------------
-
-  watchChannelForNewMessages(options: {channel: ChatChannel, onMessages: MessagesHandler}) {
-    const {channel, onMessages} = options
-    const key = chatChannelToKey(channel)
-    if (this._channelWatchers.has(key)) {
-      throw new Error(`already watching the channel ${JSON.stringify(channel)}`)
-    } else {
-      this._channelWatchers.set(key, new ChannelWatcher({channel, onMessages, bot: this}))
-    }
+  /**
+   * Listens for new chat messages on a specified channel. The `onMessage` function is called for every message your bot receives.
+   * @param onMessage
+   * @memberof Bot
+   */
+  watchChannelForNewMessages(channel: ChatChannel, onMessage: (message: ChatReadMessage) => void) {
+    keybaseExec(['chat', 'api-listen'], {
+      onStdOut: chunk => {
+        try {
+          const streamString = chunk.toString('utf8')
+          const messageObject = JSON.parse(streamString)
+          if (channel.name === messageObject.msg.channel.name) {
```

Comment:
Not super happy with how we filter messages by channel, since we don't match impteam names that aren't in the canonical order. I think this is fine for now, but requires a larger discussion.